### PR TITLE
pandas-lag-plotting

### PR DESCRIPTION
**Pull request by hassan-jvd90**
Added pandas style lag plotting for a single lag as well as a list of lags

#### Reference Issues/PRs

Pandas-style lag plotting for sktime #588
https://github.com/alan-turing-institute/sktime/issues/588 

#### What does this implement/fix? Explain your changes.

A utility for plotting lagged versions of a time series.
It allows plotting several such lagged versions together using 
the pandas lag_plot api which natively allows only one plot at
a particular lag.  

#### Does your contribution introduce a new dependency? If yes, which one?

None, that I can think of.

#### Any other comments?

Here is a demo of this utility in a collaboratory notebook: https://colab.research.google.com/drive/1tbHVqTZwGkeRjunvqGV-6Ql7NLjVJJvs?usp=sharing 

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by hassan-jvd90** [comment]
As for testing the utility, I am not exactly sure whether unit tests are needed as its a wrapper around pandas with some additions to the display. 
As a test, you can look at this notebook here: https://colab.research.google.com/drive/1tbHVqTZwGkeRjunvqGV-6Ql7NLjVJJvs?usp=sharing

**Comment by mloning** [comment]
Hi @hassan-jvd90, I suggest we create a new notebooks for time series EDA in `example/exploratory_data_analysis.ipynb` and include a simple example with your functions, that would be run automatically on CI so we have some sort of unit test for it. 

If you're interested in adding more functionality, here's another notebook with some functionality we could bring into sktime: https://github.com/mloning/intro-to-ml-with-time-series-DSSGx-2020/blob/master/notebooks/02_exploratory_data_analysis.ipynb

**Comment by mloning** [comment]
@hassan-jvd90 let me know if you need help with anything, would love to merge this PR!

**Comment by hassan-jvd90** [comment]
> 
> 
> @hassan-jvd90 let me know if you need help with anything, would love to merge this PR!

@mloning Hi! sorry for this late response. I got busy with some work-related stuff. Thanks for your encouraging follow-up. I'd love to contribute further to this project. 

As suggested by you, I have added a notebook titled exploratory_data_analysis with an example of this lag plotting utility with some explanation as well. 

I have seen the exploratory data analysis notebook shared by you as well, and I think based on that we can pull in the stuff from there as well.

Let me know what you think. Thanks!

**Review by mloning — COMMENTED** [review]
Hi @hassan-jvd90, sorry for the slow reply, this looks good in general. 

FYI there's a related PR #732.

I'd be happy to merge this as it is, but perhaps we can add a note at the top of the notebook to indicate that this is work in progress?

File: sktime/utils/plotting.py [hunk 1]
```diff
@@ -109,3 +109,70 @@ def format_fn(tick_val, tick_pos):
         ax.legend()
 
     return fig, ax
+
+
+def plot_lags(series, lags=1):
+    """Plot one or more lagged versions of a time series
```
Comment (RNKuhns) [inline sktime/utils/plotting.py:115]: Will need to make minor formatting changes to the docstring now that sktime has turned on pydocstyle docstring code quality checks.

File: sktime/utils/plotting.py [hunk 2]
```diff
@@ -109,3 +109,70 @@ def format_fn(tick_val, tick_pos):
...
+    length = 0
+    is_lag_one = isinstance(lags, int)
+
+    if is_lag_one:
+        length = 1
+        lags = [lags]
+    else:
+        length = len(lags)
+
+    fig = plt.figure(figsize=(8, 6 * length))
```
Comment (RNKuhns) [inline sktime/utils/plotting.py:166]: Rather iteratively adding the subplots, why not just make them up front (you know how many there will be). 

Also, I think the subplot layout should have multiple columns so the plot does not get too long. 

How about something like:

```python
n_cols = 3
n_rows = int(length / n_cols) + 1
fig, ax = plt.subplots(
    nrows = n_rows, ncols = n_cols, figsize=(..., ....), sharex=True, sharey=True
)
axes = ax.ravel()
```

That would give you the `fig, axes` objects you'll return later and you can just loop over the lags and for each `i, val` do the plotting using something like `pd.plotting.lag_plot(series, lag=val, ax=ax[i])`

Note that I am just using 3 columns as the example, might look fine to have 4 columns in the plotting output.

File: sktime/utils/plotting.py [hunk 3]
```diff
@@ -109,3 +109,70 @@ def format_fn(tick_val, tick_pos):
...
+    -------
+
+    fig   :  plt.Figure
+    axes  :  plt.Axis or ndarray of plt.Axis objects
+
+    Example
+    -------
+
+    Given the following time series
+
```
Comment (RNKuhns) [inline sktime/utils/plotting.py:136]: Need to update example to pass doctests. I'd also use an example dataset built into sktime (look into `load_airline()`).

File: sktime/utils/plotting.py [hunk 4]
```diff
@@ -109,3 +109,70 @@ def format_fn(tick_val, tick_pos):
...
+     the lags specified in the
+
+        >>> plot_lags(s, lags=[1,2,3]) #plots of y(t) with y(t+1),y(t+2)..
+
+    """
+    _check_soft_dependencies("matplotlib")
+    import pandas as pd
+    import numpy as np
+    import matplotlib.pyplot as plt
+
```
Comment (RNKuhns) [inline sktime/utils/plotting.py:156]: Add check of input series (e.g. `check_y()`) like used in plot_series.

**Comment by RNKuhns** [comment]
@hassan-jvd90 wanted to check-in with you on this PR. Let me know if there is anything I can do to help -- I think your close and we can get this ready to go relatively quickly.

**Comment by RNKuhns** [comment]
@hassan-jvd90 do you think you'll have time to wrap up this addition? If not, I can go ahead and finish up the suggested edits so we can merge this.

**Comment by RNKuhns** [comment]
I'm closing this and opening #1330 to finish adding this functionality.

**Pull request closed without merging**